new Promise((resolve, reject) => {
  setTimeout(() => reject("Rechaza! ðŸ”´"), 3000); // Rechaza la promesa despuÃ©s de 3 segundos
})
  .then((x) => console.log(x)) // No se ejecutarÃ¡ porque la promesa es rechazada
  .catch((e) => console.error(e)); // Captura y maneja el rechazo, imprime "2" en la consola

// Crear una nueva promesa
const promesaBasica = new Promise((resolve, reject) => {
  const exito = true; // Cambia a false para probar el caso de rechazo

  if (exito) {
    resolve("La operaciÃ³n fue exitosa!"); // Resuelve la promesa con un mensaje de Ã©xito
  } else {
    reject("Hubo un error en la operaciÃ³n."); // Rechaza la promesa con un mensaje de error
  }
});

// Manejar la promesa
promesaBasica
  .then((mensajeExito) => {
    console.log(mensajeExito); // Maneja la resoluciÃ³n de la promesa
  })
  .catch((mensajeError) => {
    console.error(mensajeError); // Maneja el rechazo de la promesa
  });
